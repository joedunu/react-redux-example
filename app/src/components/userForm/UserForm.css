.user-form {
  background-color: white;
  padding: 40px;
  & input {
    padding: 10px;
    margin-left: 10px;
  }
  .first-name {
    background-color: aqua;
  }
  & span {
    padding: 10px;
  }
}