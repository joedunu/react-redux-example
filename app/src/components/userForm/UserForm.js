'use strict'
// @flow
import React from 'react'
import PropTypes from 'prop-types'
import { Field, reduxForm } from 'redux-form'
import Paper from 'material-ui/Paper'
import Typography from 'material-ui/Typography'
import Button from 'material-ui/Button'
import Delete from 'material-ui-icons/Delete'

import './UserForm.css'
import ReduxTextField from '../common/ReduxTextField/ReduxTextField'

const onSubmitFail = (data) => {
  console.log('onSubmitFail: ', data)
}

const validate = values => {
  let errors = {}
  if (!values.firstName) {
    errors.firstName = 'First name is required'
  } else if (values.firstName.length < 2) {
    errors.firstName = 'Must be 2 characters or more'
  }
  if (!values.lastName) {
    errors.lastName = 'Last name is required'
  } else if (values.lastName.length < 2) {
    errors.lastName = 'Must be 2 characters or more'
  }
  if (!values.email) {
    errors.email = 'Email is required'
  } else if (values.email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
    errors.email = 'Invalid email address'
  }
  if (!values.mobile) {
    errors.mobile = 'Mobile number is required'
  } else if (isNaN(Number(values.mobile))) {
    errors.mobile = 'Must be a number'
  }
  return errors
}

let UserForm = (props) => {
  const {handleSubmit, enableEdit, title, classes, deleteUser, goTo} = props
  return (
    <Paper elevattion={4} className='user-form'>
      <Typography type='display1' gutterBottom>
        {title}
      </Typography>
      <form onSubmit={handleSubmit}>
        <div className='row'>
          <div className='col-4'>
            <Field
              name='firstName'
              component={ReduxTextField}
              label='First Name'
            />
          </div>
          <div className='col-2'>&nbsp;</div>
          <div className='col-4'>
            <Field name='lastName' component={ReduxTextField} type='text' label='Last Name'/>
          </div>
        </div>
        <div className='row'>
          <div className='col-4'>
            <Field name='email' component={ReduxTextField} type='text' label='Email'/>
          </div>
          <div className='col-2'>&nbsp;</div>

          <div className='col-4'>
            <Field name='mobile' component={ReduxTextField} type='text' label='Mobile'/>
          </div>
        </div>
        <div className='row'>
          <div className='col-12'>&nbsp;</div>
        </div>
        <div className='row'>
          <div className='col-6'>
            <Button className='' onClick={() => goTo('/')}>Cancel</Button>
          </div>
          <div className='col-6'>
            <Button raised color='primary' className='' type='submit'>{enableEdit ? 'Save Update' : 'Sign Up'}</Button>
            {enableEdit ? <Button raised color='accent' onClick={deleteUser}>Delete&nbsp;<Delete /></Button> : ''}
          </div>
        </div>
      </form>
    </Paper>
  )
}

UserForm.propTypes = {
  handleSubmit: PropTypes.func,
  enableEdit: PropTypes.bool,
  title: PropTypes.string.isRequired,
  deleteUser: PropTypes.func,
  classes: PropTypes.object,
  goTo: PropTypes.func.isRequired,
  formErrors: PropTypes.object.isRequired
}

export default reduxForm({
  form: 'editUser',
  validate,
  enableReinitialize: true,
  onSubmitFail
})(UserForm)
