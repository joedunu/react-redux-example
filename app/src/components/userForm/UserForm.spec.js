import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import UserForm from './UserForm'

describe('UserForm', () => {
  test('snapshot test - no props', () => {
    const component = shallow(<UserForm />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })

  test('snapshot test - basic props', () => {
    const component = shallow(
      <UserForm
        handleSubmit={jest.fn()}
        enableEdit={false}
        title='Some Form'
        goTo={jest.fn()}
      />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })
})
