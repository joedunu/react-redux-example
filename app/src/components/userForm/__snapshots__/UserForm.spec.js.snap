// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm snapshot test - basic props 1`] = `
<Connect(Form(UserForm))
  destroyOnUnmount={true}
  enableEdit={false}
  enableReinitialize={true}
  forceUnregisterOnUnmount={false}
  form="editUser"
  getFormState={[Function]}
  goTo={[Function]}
  handleSubmit={[Function]}
  keepDirtyOnReinitialize={false}
  onSubmitFail={[Function]}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  title="Some Form"
  touchOnBlur={true}
  touchOnChange={false}
  validate={[Function]}
/>
`;

exports[`UserForm snapshot test - no props 1`] = `
<Connect(Form(UserForm))
  destroyOnUnmount={true}
  enableReinitialize={true}
  forceUnregisterOnUnmount={false}
  form="editUser"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  onSubmitFail={[Function]}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
  validate={[Function]}
/>
`;
